src/request-pipeline/destination-request/index.js

```diff
@@ -33,41 +32,60 @@ export default class DestinationRequest extends EventEmitter {
         this._send();
     }
 
-    _send () {
+    _send (waitData) {
         connectionResetGuard(() => {
             var timeout = this.opts.isXhr ? DestinationRequest.XHR_TIMEOUT : DestinationRequest.TIMEOUT;
 
-            this.req = this.protocolInterface.request(this.opts);
+            this.req = this.protocolInterface.request(this.opts, res => {
+                if (waitData) {
+                    res.on('data', () => void 0);
+                    res.on('end', () => this._onResponse(res));
+                }
+            });
+
+            if (!waitData)
+                this.req.on('response', res => this._onResponse(res));
 
-            this.req.on('response', res => this._onResponse(res));
             this.req.on('error', err => this._onError(err));
             this.req.setTimeout(timeout, () => this._onTimeout());
-
             this.req.write(this.opts.body);
             this.req.end();
         });
     }
 
-    _onResponse (res) {
-        this.hasResponse = true;
+    _canTryToAuthorize (res) {
```

Comment:
Got ti now. Maybe move status code check to this function and rename it to `_shouldResendWithCredentials`?